util/unifyfs/src/unifyfs-rm.c

```diff
@@ -244,6 +248,99 @@ static int wait_server_initialization(unifyfs_resource_t* resource,
     return ret;
 }
 
+/**
+ * @brief wait until data stage operation finishes
+ *
+ * @param resource
+ * @param args
+ *
+ * @return
+ */
+
+enum {
+    UNIFYFS_STAGE_IN = 0,
+    UNIFYFS_STAGE_OUT = 1,
+};
+
+static inline unsigned int estimate_timeout(const char* manifest_file, int op)
+{
+    /* FIXME: we just wait for 20 mins.
+     * In fact, we can roughly estimate time for stage in by looking at the
+     * total file size. However, such a method is not possible for stage out
+     * because we have no idea on what the total transfer size is.
+     */
+    return 20*60;
+}
+
+static
+int wait_stage(unifyfs_resource_t* resource, unifyfs_args_t* args, int op)
```

Comment:
Document this function

Reply:
This was the floating comment above

Reply:
Sorry for this late comment. I've been thinking a bit about this. The case is different here from spawning unifyfsd because the data transfer will be terminated, while unifyfsd will continue to run. So, how about simply launching the unifyfs-stage using popen instead of using exec? Maybe, we can address this later after this PR is merged.